### PR TITLE
Bug 1636317 - Add metadata schema for Pioneer ingestion

### PR DESCRIPTION
**Pull request by acmiyaguchi**
[bug link](https://bugzilla.mozilla.org/show_bug.cgi?id=1636317)

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] Scan the PR and verify that no changes (particularly to `.circleci/config.yml`) will cause environment variables (particularly credentials) to be exposed in test logs
- [ ] If the PR comes from a fork, trigger the `integration` CI test by pushing this revision [as discussed in the README](https://github.com/mozilla-services/mozilla-pipeline-schemas#packaging-and-integration-tests-optional) and review the report posted in the comments.

### CODE REVIEW

File: templates/metadata/pioneer-ingestion/pioneer-ingestion.1.schema.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            "type": "string"
+        },
+        "studyName": {
+            "description": "Name of a particular study. Usually the addon_id.",
+            "maxLength": 100,
+            "minLength": 1,
+            "pattern": "^\\S+$",
+            "type": "string"
+        },
+        @METADATA_INGESTIONTOPLEVEL_1_JSON@,
```
Comment (acmiyaguchi) [inline templates/metadata/pioneer-ingestion/pioneer-ingestion.1.schema.json:19]: I'm not sure how many of these top level fields will actually be available to Pioneer studies. Does it make sense to prune the top-level fields based on what is actually available?

Comment (whd) [reply templates/metadata/pioneer-ingestion/pioneer-ingestion.1.schema.json:19]: Here's `metadata.attributes` for one of the decoded sample pings from today:

```
app_build_id=20200506110947
app_name=Firefox
app_update_channel=default
app_version=78.0a1
args=v=4
content_length=616
date=Thu, 07 May 2020 20:52:43 GMT
document_id=7304fcb0-9ff2-3845-969d-ef1b9f71a8bc
document_namespace=pioneer-debug
document_type=debug
document_version=1
geo_city=Blah
geo_country=Blah
geo_db_version=2020-05-04T13:02:56Z
geo_subdivision1=CA
host=upstream_docker
method=POST
normalized_app_name=Firefox
normalized_channel=Other
normalized_country_code=US
protocol=HTTP/1.0
submission_timestamp=2020-05-07T20:52:43.442758Z
uri=/submit/telemetry/7304fcb0-9ff2-3845-969d-ef1b9f71a8bc/pioneer-study/Firefox/78.0a1/default/20200506110947
user_agent_browser=Firefox
user_agent_os=Macintosh
user_agent_version=78
```

File: templates/metadata/pioneer-ingestion/pioneer-ingestion.1.schema.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "id": "http://jsonschema.net",
+    "$comment": "Added in bug 1636317; a superset of structured-ingestion for Pioneer envelope metadata.",
```
Comment (jklukas) [inline templates/metadata/pioneer-ingestion/pioneer-ingestion.1.schema.json:4]:
```suggestion
    "$comment": "Added in bug 1636317; extends the metadata included in structured ingestion, accounting for additional fields in the Pioneer envelope.",
```

File: templates/metadata/pioneer-ingestion/pioneer-ingestion.1.schema.json (hunk 1 above)
Comment (acmiyaguchi) [reply templates/metadata/pioneer-ingestion/pioneer-ingestion.1.schema.json:19]: It looks like things the operating system will be missing. And based on my reading of [`ParsePayload`](https://github.com/mozilla/gcp-ingestion/blob/9da55f6bf2ee1dc8b2844f51041bf762fbbd0edb/ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/ParsePayload.java#L191-L248), fields like the normalized_os and normalized_channel will add no information gain to the table. But changing the metadata schema at this point would require dropping any existing tables (if they exist at the moment).

Comment (whd) [reply templates/metadata/pioneer-ingestion/pioneer-ingestion.1.schema.json:19]: They don't exist (though they are about to) but dropping and recreating them is perfectly fine at this stage.

**Pull request merged**